spring-session-core/src/main/java/org/springframework/session/web/http/CookieSerializer.java

```diff
@@ -105,5 +111,24 @@ public HttpServletResponse getResponse() {
 		public String getCookieValue() {
 			return this.cookieValue;
 		}
+
+		/**
+		 * Get the cookie max age. The default is -1 which signals to delete the cookie
+		 * when the browser is closed, or 0 if cookie value is empty.
+		 * @return the cookie max age
+		 */
+		public int getCookieMaxAge() {
+			return "".equals(this.cookieValue) ? 0 : this.cookieMaxAge;
```

Comment:
I don't like this because it means we cannot have an empty String value. Could the code that sets the cookie value to "" set the max age too?